python/cugraph/cugraph/experimental/dask/centrality/katz_centrality.py

```diff
@@ -0,0 +1,155 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dask.distributed import wait, default_client
+from cugraph.dask.common.input_utils import (get_distributed_data,
+                                             get_vertex_partition_offsets)
+from pylibcugraph.experimental import (ResourceHandle,
+                                       GraphProperties,
+                                       MGGraph,
+                                       katz_centrality as pylibcugraph_katz
+                                       )
+import cugraph.dask.comms.comms as Comms
+import dask_cudf
+import cudf
+import cupy
+
+
+def call_katz_centrality(sID,
+                         data,
+                         graph_properties,
+                         store_transposed,
+                         do_expensive_check,
+                         src_col_name,
+                         dst_col_name,
+                         num_verts,
+                         num_edges,
+                         vertex_partition_offsets,
+                         aggregate_segment_offsets,
+                         alpha,
+                         beta,
+                         max_iter,
+                         tol,
+                         nstart,
+                         normalized):
+    handle = Comms.get_handle(sID)
+    h = ResourceHandle(handle.getHandle())
+    srcs = data[0][src_col_name]
+    dsts = data[0][dst_col_name]
+    weights = cudf.Series(cupy.ones(srcs.size, dtype="float32"))
+
+    if "value" in data[0].columns:
+        weights = data[0]['value']
+
+    initial_hubs_guess_values = None
+    if nstart:
+        initial_hubs_guess_values = nstart["values"]
+
+    mg = MGGraph(h,
+                 graph_properties,
+                 srcs,
+                 dsts,
+                 weights,
+                 store_transposed,
+                 num_edges,
+                 do_expensive_check)
+
+    result = pylibcugraph_katz(h,
+                               mg,
+                               initial_hubs_guess_values,
+                               alpha,
+                               beta,
+                               tol,
+                               max_iter,
+                               do_expensive_check)
+    return result
+
+
+def convert_to_cudf(cp_arrays):
+    """
+    create a cudf DataFrame from cupy arrays
+    """
+    cupy_vertices, cupy_values = cp_arrays
+    df = cudf.DataFrame()
+    df["vertex"] = cupy_vertices
+    df["katz_centrality"] = cupy_values
+    return df
+
+
+def EXPERIMENTAL__katz_centrality(
+    input_graph, alpha=None, beta=1.0, max_iter=100, tol=1.0e-6,
+    nstart=None, normalized=True
```

Comment:
regarding the `nstart` parameter, I don't see it being handled properly (case where the dataframe was renumbered, `nstart` vertices should have the internal vertex ids)